crypto/mem.c

```diff
@@ -164,9 +165,17 @@ static int shouldfail(void)
 void ossl_malloc_setup_failures(void)
 {
     const char *cp = getenv("OPENSSL_MALLOC_FAILURES");
-
-    if (cp != NULL && (md_failstring = strdup(cp)) != NULL)
-        parseit();
+    size_t cplen = 0;
+
+    if (cp != NULL) {
+        /* if the value is too long we'll just ignore it */
+        cplen = strlen(cp);
+        if (cplen <= CRYPTO_MEM_CHECK_MAX_FS) {
```

Comment:
(minor) It's probably safer (in terms of future changes to this code) to use `sizeof(md_failbuf) - 1` on both this line and the one below

Reply:
will do

Reply:
> Got some windows CI issue with that change so reverted it

What was the issue? I can't think why it would cause a failure (I didn't get to see this before the next run hid the logs, and I don't know how to find the CI logs for an old version of a PR)